### DISCUSSION TITLE
Possible to have seperate modifiers for placeholder and main image? SwiftUI

### DISCUSSION THREAD

**Discussion by brzzdev** [onevcat/Kingfisher#2013] [Q&A]
Currently, this is my code:
```
KFImage(url)
    .placeholder {
        Image(systemName: "questionmark.circle.fill")
            .resizable()
            .aspectRatio(contentMode: .fit)
            .padding(.horizontal, 8)
    }
    .resizable()
    .aspectRatio(contentMode: .fill)
    .frame(width: 98, height: 128)
    .modifier(RoundedEdge(width: 2, color: color ?? .purple, cornerRadius: 6))
    .clipped()
```

It seems it's impossible to make the placeholder .fit or respect the padding because it's immediately overridden by the .fill

Apple's AsyncImage I can do:
```
AsyncImage(url: url) { image in
    image
        .resizable()
        .aspectRatio(contentMode: .fill)
} placeholder: {
    Image(systemName: .questionmarkCircleFill)
        .resizable()
        .aspectRatio(contentMode: .fit)
        .padding(.horizontal, 8)
}
```

but it doesn't look like I can with KFImage? I looked at onSuccess and onFailureImage but looks like they don't quite do what I want.

In UIKit I used to kf.setImage and switch on the result

Am I missing something obvious or is this not possible using KFImage? Thanks

**Comment by onevcat** [comment]
Maybe a duplicate to #2016 , a frame setup is necessary for changing image aspect in placeholder.

And also the new `Image -> some View` configuration can solve it, but it is not yet fully prepared and I will try to contribute to it more soon.
